core/commands/keystore.go

```diff
@@ -415,26 +426,28 @@ var keyRmCmd = &cmds.Command{
 
 func verifyIDFormatLabel(formatLabel string) error {
 	switch formatLabel {
-	case "b58mh":
-		return nil
-	case "b36cid":
+	case "b58mh", "v0":
 		return nil
+	default:
+		_, err := mbase.EncoderByName(formatLabel)
+		return err
 	}
-	return fmt.Errorf("invalid output format option")
 }
 
 func formatID(id peer.ID, formatLabel string) string {
 	switch formatLabel {
-	case "b58mh":
+	case "b58mh", "v0":
 		return id.Pretty()
-	case "b36cid":
-		if s, err := peer.ToCid(id).StringOfBase(mbase.Base36); err != nil {
+	default:
+		enc, err := mbase.EncoderByName(formatLabel)
+		if err != nil {
+			panic("invalid IPNS key encoding")
```

Comment:
Separating the validation function from the formatting function feels like a footgun. As we may accidentally call `formatID` without calling `verifyIDFormatLabel` and ending up with a panic. 

How about creating a tiny struct to combine these two functions, one for creating the encoder and the other for running it.